test/libsolidity/SemanticTest.h

```diff
@@ -30,6 +30,13 @@
 namespace solidity::frontend::test
 {
 
+struct EventInformation
```

Comment:
How about this?
```suggestion
struct AnnotatedEventSignature
```

Also looks like you're not using `nonIndexedTypes` anywhere and only the size of `indexedTypes` is being checked by an assert.

I'd also rename `findEvent()` to match.

Reply:
How about renaming `matchEvent()` to `matchNonAnonymousEvent()`? I think that the fact that it will never return an anonymous one is an important detail.